mico-core/src/main/java/io/github/ust/mico/core/REST/ServiceController.java

```diff
@@ -160,31 +160,48 @@
         if (!serviceOpt.isPresent()) {
             return ResponseEntity.notFound().build();
         }
-        MicoService service = serviceOpt.get();
-
 
         Optional<MicoService> serviceDependeeOpt = serviceRepository.findByShortNameAndVersion(newServiceDependee.getDependedService().getShortName(),
-            newServiceDependee.getDependedService().getVersion().toString());
+            newServiceDependee.getDependedService().getVersion());
 
         if (!serviceDependeeOpt.isPresent()) {
             return ResponseEntity.notFound().build();
```

Comment:
You could add a reason message to indicate to the user what is missing or wrong.